### PR TITLE
separate suitesparse components: graphBLAS, Mongoose

### PR DESCRIPTION
**Pull request by minrk**
these are shipped with suitesparse 5.4, but unlike umfpack and friends, are independently versioned and distributed as well.

see https://github.com/conda-forge/suitesparse-feedstock/issues/55

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

I've initialized maintainers of the suitesparse recipe as maintainers of these as well:

- @grlee77
- @jakirkham
- @basnijholt
- @jayfurmanek

pleas confirm if this is okay. If not, I'll remove you.

### CODE REVIEW

**Comment by grlee77** [comment]
It looks like the failure on OS X is related to OpenMP

Are we supposed to be using `llvm-openmp` instead of `openmp` in the `meta.yaml` for OS X?


<details>
<summary>Details from log</summary>

```
Packaging graphblas
INFO conda_build.build:build(1570): Packaging graphblas
Packaging graphblas-3.1.0-h4a8c4bd_0
INFO conda_build.build:bundle_conda(891): Packaging graphblas-3.1.0-h4a8c4bd_0
number of files: 5
Making absolute symlink lib/libgraphblas.dylib -> libgraphblas.3.dylib relative
Fixing linking of @rpath/libgraphblas.3.dylib in /usr/local/miniconda/conda-bld/graphblas_1570720521276/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libgraphblas.3.1.0.dylib
New link location is None
Fixing linking of /usr/lib/libSystem.B.dylib in /usr/local/miniconda/conda-bld/graphblas_1570720521276/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libgraphblas.3.1.0.dylib
New link location is None
Fixing linking of @rpath/libomp.dylib in /usr/local/miniconda/conda-bld/graphblas_1570720521276/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libgraphblas.3.1.0.dylib
New link location is None

  ERROR :: lib/libomp.dylib not in prefix_owners
   INFO (graphblas,lib/libgraphblas.3.1.0.dylib): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
  ERROR (graphblas,lib/libgraphblas.3.1.0.dylib): Needed DSO lib/libomp.dylib found in ['llvm-openmp']
  ERROR (graphblas,lib/libgraphblas.3.1.0.dylib): .. but ['llvm-openmp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
Traceback (most recent call last):
```

</details>

File: recipes/mongoose/disable-coverage-clang.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -323,9 +323,9 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" ST
+     # using Clang
+     SET(CMAKE_CXX_FLAGS "-O3 -fwrapv")
+     # Debug flags for Clang
+-    SET(CMAKE_CXX_FLAGS_DEBUG "--coverage -g -fwrapv")
+-    SET(CMAKE_C_FLAGS_DEBUG "--coverage -g")
+-    SET(CMAKE_EXE_LINKER_FLAGS_DEBUG "--coverage -g")
++    SET(CMAKE_CXX_FLAGS_DEBUG "-g -fwrapv")
```
Comment (isuruf) [inline recipes/mongoose/disable-coverage-clang.patch]: If these are getting used, that means you are building in Debug mode.

Comment (minrk) [reply recipes/mongoose/disable-coverage-clang.patch]: the build unconditionally does debug-compilation as well for tests, even if building a release

**Comment by minrk** [comment]
This appears to be all ready to go now

**Comment by minrk** [comment]
ping @conda-forge/staged-recipes, should be ready for merge. The corresponding SuiteSparse PR has been merged.

**Pull request merged**